python/ray/serve/experimental/gradio_visualize_graph.py

```diff
@@ -39,6 +41,73 @@ def lazy_import_gradio():
     return _gradio
 
 
+def get_block(node, **kwargs):
+    gr = lazy_import_gradio()
+
+    return_type = node.get_return_type()
+    node_name = _DAGNodeNameGenerator().get_node_name(node)
+    if return_type is None:
+        logger.warning(
+            f"Return type for {node_name} node was not provided. "
+            "Defaulting to gr.Textbox."
+        )
+        return gr.Textbox(**kwargs)
+
+    if return_type == "int":
+        return gr.Number(precision=0, **kwargs)
+    elif return_type == "float":
+        return gr.Number(**kwargs)
+    elif return_type == "str":
+        return gr.Textbox(**kwargs)
+    elif return_type == "bool":
+        return gr.Checkbox(**kwargs)
+    elif return_type == "pandas.core.frame.DataFrame":
+        return gr.Dataframe(**kwargs)
+    elif (
+        return_type == "list"
+        or return_type == "dict"
+        or return_type == "typing.List"
+        or return_type == "typing.Dict"
+    ):
+        return gr.JSON(**kwargs)
+    else:
+        from PIL import ImageFile
+
+        if (
+            issubclass(locate(return_type), ImageFile.ImageFile)
+            or return_type == "numpy.ndarray"
+        ):
+            return gr.Image(**kwargs)
+
+    logger.warning(
```

Comment:
I don't think this `warning` (or any of the code below it) ever executes. I believe the `else` clause above will catch all the lingering options.

Reply:
I think it will if `issubclass(locate(return_type), ImageFile.ImageFile) or return_type == "numpy.ndarray"` evaluates to false. Then the function will not return and exits the `else` clause.

Reply:
Oh, that makes sense.